Update Hist Funct with new parameters

nkorinek opened the pull request:
Added in the range argument and I fixed the colors argument as well. I feel like my range argument is added in a bit sloppily, I imagine there is a better way to do that. @lwasser any ideas? Also, the first test for range works with a one dimensional array, but the multi dimensional test for range fails. I can't use the same method I used for the one dimensional array, and was wondering if there was a good way to test that the values were modified. @lwasser advice here would also be appreciated. This isn't done yet, but it's closer then it was!

nkorinek commented:
I meant to put this into the existing pull request, apologies!

lwasser commented on earthpy/plot.py:
```diff
@@ -409,7 +409,7 @@ def hist(
     xlabel="",
     ylabel="",
-    **kwargs,
+    hist_range=[],
```
This is the correct PR design @nkorinek as a side note...  `range` is a tuple in ax.hist. so you should call it range and then accept a tuple as the input value.

lwasser commented on earthpy/plot.py:
```diff
@@ -431,12 +431,14 @@ def hist(
-        These will be passed to the matplotlib .hist call
-        See https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.hist.html
+        The text to print on the y axis.
+    range : list (default = [np.nanmin(arr), np.nanmax(arr)])
```
this should be a tuple.https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.axes.Axes.hist.html

nkorinek replied:
I borrowed some of the documentation from matplotlib to document the range parameter in EarthPy. Is there a way to cite them, or is that not common practice?

lwasser replied:
i've seen people say: from the docs and provide a link? or you can just provide a link as i see was in the code previously?? does that help?

lwasser commented on earthpy/plot.py:
```diff
@@ -431,12 +431,14 @@ def hist(
-        See https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.hist.html
+        The text to print on the y axis.
+    range : list (default = [np.nanmin(arr), np.nanmax(arr)])
+        The values over which the histogram will range. Should be
```
follow the matplotlib docs but then say we will calculate it they don't provide it using the min and max as you did

lwasser commented on earthpy/plot.py:
```diff
@@ -497,8 +501,14 @@ def hist(
                 the_color = colors[0]
             else:
                 the_color = colors[i]
+            if not hist_range:
```
this should be called `range()` not hist_range

lwasser replied:
ok let's keep this as `hist_range` after talking!!

lwasser commented on earthpy/plot.py:
```diff
@@ -497,8 +501,14 @@ def hist(
             else:
                 the_color = colors[i]
+            if not hist_range:
+                hist_range = [np.nanmin(band), np.nanmax(band)]
```
this should be a tuple not a list `range = (np.nanmin(arr), np.nanmax(arr))` and also should use the input array. you only have to calculate range once for the entire array regardless of single vs multiple band. if there are many bands, please calculate the range on the ENTIRE ARRAY.  so it's not in the loop

lwasser commented on earthpy/plot.py:
```diff
@@ -497,8 +501,14 @@ def hist(
+                bins=bins,
+                color=the_color,
+                alpha=0.8,
+                range=hist_range,
```
hist range should just be range

lwasser commented on earthpy/plot.py:
```diff
@@ -521,15 +531,10 @@ def hist(
-            color=colors[0],
-            **kwargs,
-        )
+        ax.hist(arr_comp, range=hist_range, bins=bins, color=colors[0])
```
range rather than hist_Range

lwasser commented on earthpy/tests/test_hist.py:
```diff
@@ -147,10 +160,8 @@ def test_hist_masked_array(image_array_2bands):
-    for an_axis in ax:
-        assert len(ax[0].patches) == nbins
-
+    for a in ax:
```
does this run without the `[0]` ?

lwasser commented:
@nkorinek this is actually adding to my pr which is what i wanted you to do. the other PR tries to merge to master which is incorrect given what we are working on together here!

nkorinek commented:
@lwasser I added in the tests that we talked about. I also updated the function bc while I was writing the xlabel and ylabel test I realized they were never actually assigned in the function. That should be all cleared up now! I added the x and y label a bit messily, but I couldn't think of a better way to do it. For the title parameter, there was a check to see if the parameter exists. I did the same for the x and y label. I think that it would also work without checking if the x and y labels are there or not, and just running `ax.set(xlabel=xlabel, ylabel=ylabel)`, but I followed along with what was already in the script. Everything works now and is tested, so hopefully this is good to merge!
